### ISSUE TITLE
libretro core : crash when starting JAP games

### ISSUE THREAD

**Issue by bigboo3000** [inolen/redream#150]
redream libretro core crash after 2s (120 frames) when loading japanese games.
I tested with retroarh 1.6.7, latest redream core, on windows 10 x64 and OPENGL on radeon card.
some game tested:
Dead or alive 2 special edition
Space channel part 2
Guilty Gear X

They work fine with standalone redream x64 on Windows 10.

**Comment by bigboo3000** [comment]
problem fixed:
for retroacch boot.bin must be in \system and not in system\dc\ (like reicast).
So it was using HLE bios.
HLE bios seems to have problems with japanese games.

**Comment by inolen** [comment]
You're right on system/ vs system/dc. It's supposed to be system/dc/, but:
https://github.com/inolen/redream/blob/master/src/host/retro_host.c#L204

It calls fs_set_appdir with sysdir, not the newly formatted appdir.

**Comment by inolen** [comment]
Pushed a quick fix for that:
https://github.com/inolen/redream/commit/5feae48f5fec62d1c1535b75024dad05c034734b

**Comment by bigboo3000** [comment]
Thanks.
I change the issue title because the real problem is with HLE BIOS.

**Comment by bigboo3000** [comment]
I checked again and in fact there are different problems for japanese games with hle bios:

Border Down, Ikarouga, Karous, Radirgy, Space Channel 5 Part 2, Triggerheart Exelica, Under Defeat:
crash with this error in log:
holly_maple_dma unhandled pattern 0x3

Cosmic Smash:
some kind of reset loop in the sega licence screen

Guilty Gear X:
the fight never start (no "fight!" animation/introduction) and fighters are like asleep (very strange lol)

**Comment by bigboo3000** [comment]
The only one non-japanese I found with problems is sf rush 2049 PAL:
With hle bios no controller is detected
![image](https://user-images.githubusercontent.com/16479698/32884841-f146e284-cabb-11e7-93f2-7851cc4970c8.png)

**Comment by bigboo3000** [comment]
with latest version sf rush 2049 now crash just after boot in hle (still works with real bios):
[C:\projects\redream\src\guest\pvr\ta.c:385] Check failed: !ctx->rendering

**Comment by inolen** [comment]
Adding Confidential Mission to the list of games broken with the HLE bios.

**Comment by inolen** [comment]
As of latest, SF Rush now has problems because it internally checksums the bios, setting some flag which makes the game stop working after the main menu if it fails. Since there is no bios, this checksum test will never pass.

To solve this, we'll have to reintroduce the per-game patching code I was tinkering with when I did the widescreen hack work.

Notes for future me: checksum test is called at 0x8c08bc9a, checksum test starts at 0x8c0727c8, checksum result flag is stored at 0x8c0cfe00.

**Comment by 9madsheep** [comment]
From japanese_cake:
https://assemblergames.com/threads/new-custom-dreamcast-bios-v1-032-relased.62035/
____________________________________________________________________________________________________
"San Francisco RUSH 2049" checks whether the system video broadcast is "correct": the
NTSC/US and PAL/EURO version expect the video broadcast to be set to NTSC and PAL,
respectively

So far I have identified 3 games that do such check:
- Elemental Gimmick Gear v1.001 (1999)(Vatical)(NTSC)(US)[!] (T41601N)
- San Francisco RUSH 2049 v1.001 (2000)(Midway)(NTSC)(US)[!] (T-9707N)
- San Francisco RUSH 2049 v1.002 (2000)(Midway)(PAL)(M6)[!] (T-9709D-50)
_____________________________________________________________________________________________________

also some other games have protections like
El Dorado Gate series from Capcom

**Comment by inolen** [comment]
All of the games mentioned except SF Rush should now boot with the HLE bios as of d27198606420d1559e6c06384f74e90db09d3da4.

**Comment by bigboo3000** [comment]
Confirmed. Do you want to keep this issue open for SF Rush?

**Comment by bigboo3000** [comment]
I found other games not working with hle (hang after sega licence screen):
Fur Fighters and Metropolis Street Racer:
unsupported GDC_READ
unsupported GDC_READ
unsupported GDC_READ
unsupported GDC_READ
unsupported GDC_READ
unsupported GDC_READ
unsupported GDC_READ
unsupported GDC_READ
unsupported GDC_READ
unsupported GDC_READ
(...)

Floigan Bros/NBA2K2/Ooga Booga (not working with real bios too but error is different):
unsupported GDC_READ
